### PR TITLE
Add test cases to product scenarios

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
The purpose of this PR is to add new test cases and util functions to ScenarioTestBase.

## Approach
Test case convertMalformedSoapToJson is added to 1.1.2-soap-to-json-using-messagetype-property test. 

Util function getRequestResponseList() is added to ScenarioTestBase. When dataproviders are used to extract request and response file contents this util function can be called. 

## Test environment
This is tested locally in following environment.
JDK - 1.8
OS - macOS High Sierra

### CODE REVIEW

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java [hunk 1]
```diff
@@ -37,20 +37,25 @@
...
+import java.text.Collator;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Properties;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
+import static org.wso2.esb.integration.common.utils.common.FileManager.readFile;
+
 public class ScenarioTestBase {
```
Comment (abeykoon) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Class level comment please.

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java [hunk 2]
```diff
@@ -276,17 +282,46 @@ public Boolean call() throws Exception {
...
 
-    protected String getStringFromDocument(Document doc) throws TransformerException {
-        DOMSource domSource = new DOMSource(doc);
-        StringWriter writer = new StringWriter();
-        StreamResult result = new StreamResult(writer);
-        TransformerFactory tf = TransformerFactory.newInstance();
-        Transformer transformer = tf.newTransformer();
-        transformer.transform(domSource, result);
-        return writer.toString();
+    protected List<Object[]> getRequestResponseList(String testCase) throws Exception {
```
Comment (abeykoon) [inline product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: Shall we add method comment?

Comment (dulanjalidilmi) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: This method was updated and method comment was added in PR [https://github.com/wso2/product-ei/pull/2937](https://github.com/wso2/product-ei/pull/2937)

File: product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java (hunk 1 above)
Comment (dulanjalidilmi) [reply product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java]: This is fixed by [https://github.com/wso2/product-ei/pull/2991](https://github.com/wso2/product-ei/pull/2991)

**Pull request merged**